test/index.test.ts

```diff
@@ -192,6 +192,116 @@ describe('HipThrusTS', () => {
       });
     });
   });
```

Comment:
I didn't mean for the test to look exactly like that...  That commented code was only meant to serve as more of a spec of what needs to be included
- [ ] give each call to HTPipeAttachData its own "it" block
- [ ] after each call, verify that the type of what's returned is EXACTLY what it's supposed to be - note that I'm not sure I trust that expectType does what it advertises - make sure that expectType<{a: string}>({a: "wt", b: "f"}) FAILS! ... if it doesn't fail, it's only checking if the parameter is ASSIGNABLE to the type parameter.  In that case you might need to expectType BOTH directions (hint: typeof will be your friend)
- [ ] as we discussed in chat, that ts-expect-fail thing ... btw, for that one you should make sure it isn't ACTUALLY called by putting it into a function or something.
- [ ] I think for clarity's sake, don't share the data (the pojos) ... keep the it() blocks independent and not dependent on some shared data.

Reply:
just in case: make sure that expectType<{a: string}>({a: "wt", b: "f"}) FAILS!
error: Argument of type '{ a: string; b: string; }' is not assignable to parameter of type '{ a: string; }'.
  Object literal may only specify known properties, and 'b' does not exist in type '{ a: string; }'

Reply:
I refactor all the test case to have it block
also i wasn't sure what is the best way to compare all types so i decide to build the HTPipeAttachData return type
the only problem if we will change the return type of HTPipeAttachData we need to change it in tests

Reply:
tasks from above done in HTPipeattachdata-dx-todos PR